# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.h
# Commit: f6ce0e05c92e
# Full Hash: f6ce0e05c92e16e33c4bc136228ab4bc6058f358
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P4: Cache aria group attributes and introduce Accessible getter. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134207
# ==============================================================================

diff -r afea577be988 -r f6ce0e05c92e accessible/basetypes/Accessible.h
--- a/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:35 2022 +0000
+++ b/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:35 2022 +0000
@@ -291,6 +291,15 @@
 
   virtual HyperTextAccessibleBase* AsHyperTextBase() { return nullptr; }
 
+ protected:
+  // Some abstracted group utility methods.
+
+  /**
+   * Get ARIA group attributes.
+   */
+  virtual void ARIAGroupPosition(int32_t* aLevel, int32_t* aSetSize,
+                                 int32_t* aPosInSet) const = 0;
+
  private:
   static const uint8_t kTypeBits = 6;
   static const uint8_t kGenericTypesBits = 18;